scipy/stats/_distribution_infrastructure.py

```diff
@@ -5219,3 +3582,289 @@ def __rtruediv__(self, other):
+        The corresponding probabilities of selecting each random variable.
+        Must be non-negative and sum to one.
+
+    Methods
+    -------
+    support
+
+    sample
+
+    moment
+
+    mean
+    median
+    mode
+
+    variance
+    standard_deviation
+
+    skewness
+    kurtosis
+
+    pdf
+    logpdf
+
+    cdf
+    icdf
+    ccdf
+    iccdf
+
+    logcdf
+    ilogcdf
+    logccdf
+    ilogccdf
+
+    entropy
+
+    Notes
+    -----
+    The following abbreviations are used throughout the documentation.
+
+    - PDF: probability density function
+    - CDF: cumulative distribution function
+    - CCDF: complementary CDF
+    - entropy: differential entropy
+    - log-*F*: logarithm of *F* (e.g. log-CDF)
+    - inverse *F*: inverse function of *F* (e.g. inverse CDF)
+
+    References
+    ----------
+    .. [1] Mixture distribution, *Wikipedia*,
+           https://en.wikipedia.org/wiki/Mixture_distribution
+
+    """
+    # Todo:
+    # Fix Normal(mu=0.5).logentropy() runtime warning
+    # Add support for array shapes, weights
+
+    def _input_validation(self, components, weights):
+        if len(components) == 0:
+            message = ("`components` must contain at least one random variable.")
+            raise ValueError(message)
+
+        for var in components:
+            # will generalize to other kinds of distributoins when there
+            # *are* other kinds of distributions
+            if not isinstance(var, ContinuousDistribution):
+                message = ("Each element of `components` must be an instance of "
+                           "`ContinuousDistribution`.")
+                raise ValueError(message)
+            if not var._shape == ():
+                message = "All elements of `components` must have scalar shapes."
+                raise ValueError(message)
+
+        if weights is None:
+            return
+
+        weights = np.asarray(weights)
+        if weights.shape != (len(components),):
+            message = "`components` and `weights` must have the same length."
+            raise ValueError(message)
+
+        if not np.issubdtype(weights.dtype, np.inexact):
+            message = "`weights` must have floating point dtype."
+            raise ValueError(message)
+
+        if not np.isclose(np.sum(weights), 1.0):
+            message = "`weights` must sum to 1.0."
+            raise ValueError(message)
+
+        if not np.all(weights >= 0):
+            message = "All `weights` must be non-negative."
+            raise ValueError(message)
+
+    def __init__(self, components, /, *, weights=None):
+        self._input_validation(components, weights)
+        n = len(components)
+        dtype = np.result_type(*(var._dtype for var in components))
+        self._shape = np.broadcast_shapes(*(var._shape for var in components))
+        self._dtype, self._components = dtype, components
+        self._weights = np.full(n, 1/n, dtype=dtype) if weights is None else weights
```

Comment:
Should component's and weights be public? Or if you have these private because you don't want to encourage users to modify them manually, should their be read-only properties to get them? I think there are use cases where you would want to have access to the weights and components in a function which consumes Mixture distributions.

Reply:
I don't really have an opinion, so I defaulted to private.

Reply:
I see. So when we do allow for fitting mixture models, the things you can fit are the parameters of the components and the weights, and it's important to be able check these. But having the weights and components private is similar to having distributions where the shape and other parameters are private, and it looks like that's currently the case too. I guess you didn't have an opinion on making any parameters public vs private, so defaulted to all being private? Being able to figure out the parameters of an instantiated distribution seems pretty important imo.

Reply:
To clarify, the shape parameters of instances of `ContinuousDistribution` are public, like `Normal(mu=mu).mu` because that has been requested in the past.

The broadcasted shape of a distribution with array parameters is currently private `Normal(mu=mu)._shape`.

I can make the underlying distributions and weights public if desired.

Reply:
If the parameters are modifiable but changing them does nothing, I think they're better off as read only properties.

Reply:
> I see. I got mixed up by the following:

It would be nice to raise an error (at least for now, while RVs are immutable). I think I can do that by using the `property` built-in function rather than overriding `setattr`, and that would probably be faster, too.

Reply:
Fixed by gh-21767.

Reply:
Made public properties in recent commit.